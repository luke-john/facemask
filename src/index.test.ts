import { css } from "styled-components";
import { responsiveStyles } from ".";

const scaleProperties = responsiveStyles({
  mediaQueries: ["@media(min-width: 731px)", "@media(min-width: 1695px)"],
  tokens: {
    spacing: {
      moon: [4, 6, 10],
      mars: [8, 13, 20],
      earth: [16, 26, 41],
      saturn: [32, 52, 82],
      sun: [66, 106, 180]
    },
    type: {
      snail: [13, 16, 20],
      mouse: [16, 20, 26],
      otter: [20, 26, 32],
      dolphin: [26, 32, 41],
      hippo: [32, 41, 52],
      mammoth: [41, 52, 66],
      whale: [52, 66, 83]
    }
  }
});

it("scales properties", () => {
  const scaledProperties = scaleProperties(scale => ({
    padding: scale.spacing.moon,
    "& div": {
      padding: scale.spacing.saturn
    }
  }));

  expect(
    css`
      ${scaledProperties};
    `
  ).toMatchSnapshot();
});

it("builds responsive shorthand queries", () => {
  const scaledProperties = scaleProperties((scale, { shorthand }) => ({
    padding: shorthand([scale.spacing.moon, scale.spacing.earth]),
    margin: scale.spacing.earth,
    "& div": {
      padding: scale.spacing.saturn,
      margin: shorthand([
        scale.spacing.earth,
        scale.spacing.saturn,
        scale.spacing.sun,
        scale.spacing.mars
      ])
    }
  }));

  expect(
    css`
      ${scaledProperties};
    `
  ).toMatchSnapshot();
});
