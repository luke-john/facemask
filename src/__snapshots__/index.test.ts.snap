// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builds responsive shorthand queries 1`] = `
Array [
  "
      ",
  "padding:  4 16; margin: 16; @media(min-width: 731px) {
  padding:  6 26; margin: 26;
} @media(min-width: 1695px) {
  padding:  10 41; margin: 41;
} & div {
  padding: 32; margin:  16 32 66 8; @media(min-width: 731px) {
  padding: 52; margin:  26 52 106 13;
} @media(min-width: 1695px) {
  padding: 82; margin:  41 82 180 20;
}
}",
  ";
    ",
]
`;

exports[`scales properties 1`] = `
Array [
  "
      ",
  "padding: 4; @media(min-width: 731px) {
  padding: 6;
} @media(min-width: 1695px) {
  padding: 10;
} & div {
  padding: 32; @media(min-width: 731px) {
  padding: 52;
} @media(min-width: 1695px) {
  padding: 82;
}
}",
  ";
    ",
]
`;
